include/alpaka/mem/fence/MemFenceCpu.hpp

```diff
@@ -59,10 +59,7 @@ namespace alpaka
                  *   a == 10 && b == 20
                  */
 
-                /* TODO: Once we have C++17 enabled by default the following line should be turned into the usual
-                 * `auto x = ...` notation.
-                 */
-                static std::atomic<int> dummy{42};
+                static auto dummy = std::atomic<int>{42};
```

Comment:
Does this make the code better? Is this change mandated by a style guide? I would be fine with the original version and just remove the comment.

Reply:
I think it follows the AAA aka almost always auto (which in C++17 becomrs always auto) which we have as a recommendation but not a requirement

Reply:
Looks like older clang versions don't (fully) support this feature anyway :-(

Reply:
Wait, this might be solved by the Ubuntu upgrade in #1484. The failing checks are still using the old gcc stdlib from Ubuntu 18.04.